frontend/test/__support__/ui.tsx

```diff
@@ -88,6 +88,12 @@ export function renderWithProviders(
       { sdk: createMockSdkState(), ...initialState },
       ...sdkReducerNames,
     ) as SdkStoreState;
+
+    // Enable the embedding feature by default in SDK tests, unless explicitly disabled.
+    // Without this, SDK components will not render due to missing token features.
+    if (!storeInitialState.settings && initialState.settings) {
```

Comment:
Why don't we need this before this PR? Does it mean this PR modifies how our SDK works? Or does it mean we use a different setup in this PR?

Reply:
@WiNloSt It's because we are using JWT authentication in tests without the token features (e.g. without "embedding"), so without the token feature the components would not render - because of the check in PublicComponentWrapper.

Reply:
Should this be in each test's setup instead? In this case I imagine you only need to add setup proper `token-features` in this PR's unitest's setup. Then you can leave this util as is.

There are other unit tests that set up token features in unit tests for example
https://github.com/metabase/metabase/blob/2b0dc0b0c6c13e466f71b21f2d809720045633a2/frontend/src/metabase/admin/settings/app/components/SettingsEditor/tests/embedding/setup.tsx#L34-L37

Reply:
@WiNloSt As `PublicComponentWrapper` is a part of every SDK component, I would have to modify every other SDK-related tests (e.g. `InteractiveQuestion`, `StaticQuestion`, `StaticDashboard`) and other tests that uses JWT authentication in SDK.

This only affects `mode: "sdk"` tests that does not have a `settings` object provided to it, so tests that provides their own `settings` in the initial state should still work. If this feels too implicit, I can add this to every SDK test instead.

Reply:
You have a point, let's keep it like this.